src/buf/fixed.rs

```diff
@@ -0,0 +1,184 @@
+//! Buffers pre-registered with the kernel.
+//!
+//! This module provides facilities for registering in-memory buffers with
+//! the `tokio-uring` runtime. Operations like [`File::read_fixed_at`][rfa] and
+//! [`File::write_fixed_at`][wfa] make use of buffers pre-mapped by
+//! the kernel to reduce per-I/O overhead.
+//! The [`BufRegistry::register`] method is used to register a collection of
+//! buffers with the kernel; it must be called before any of the [`FixedBuf`]
+//! handles to the collection's buffers can be used with I/O operations.
+//!
+//! [rfa]: crate::fs::File::read_fixed_at
+//! [wfa]: crate::fs::File::write_fixed_at
+
+use super::{IoBuf, IoBufMut};
+use crate::driver::{self, Buffers};
+
+use std::cell::RefCell;
+use std::io;
+use std::mem::ManuallyDrop;
+use std::rc::Rc;
+
+/// An indexed collection of I/O buffers pre-registered with the kernel.
+///
+/// `BufRegistry` allows the application to manage a collection of buffers
+/// allocated in memory, that can be registered in the current `tokio-uring`
+/// context using the [`register`] method.
+///
+/// [`register`]: Self::register
+///
+/// A `BufRegistry` value is a lightweight handle for a collection of
+/// allocated buffers. Cloning of a `BufRegistry` creates a new reference to
+/// the same collection of buffers.
+///
+/// The buffers of the collection are not deallocated until:
+/// - all `BufRegistry` references to the collection have been dropped;
+/// - all [`FixedBuf`] handles to individual buffers in the collection have
+///   been dropped, including the buffer handles owned by any I/O operations
+///   in flight;
+/// - every `tokio-uring` runtime the buffers have been registered in
+///   has terminated.
+#[derive(Clone)]
+pub struct BufRegistry {
```

Comment:
The path in the public API is already `buf::fixed::BufRegistry`, and I'm of the school of thought that [instructs](https://github.com/rust-lang/rfcs/blob/master/text/0356-no-module-prefixes.md) to use the module names in preference to prefixes. Then again, I've got `buf::fixed::FixedBuf`, but that has not been named `Buf` to avoid clashes with `bytes::Buf`.

Perhaps we could justify the prefix as describing the distinctive attribute of these objects, so `FixedBufRegistry` it should be.